std/json.d

```diff
@@ -346,10 +346,11 @@ struct JSONValue
             type_tag = JSON_TYPE.STRING;
             store.str = arg;
         }
-        else static if(is(T : char[]))
+        else static if(isNarrowString!T)
```

Comment:
```isSomeString``` might not be good either: "Static arrays of characters (like char[80]) are not considered built-in string types."